helm-app-operator/hack/tests/e2e-helm.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+#===================================================================
+# FUNCTION trap_add ()
+#
+# Purpose:  prepends a command to a trap
+#
+# - 1st arg:  code to add
+# - remaining args:  names of traps to modify
+#
+# Example:  trap_add 'echo "in trap DEBUG"' DEBUG
+#
+# See: http://stackoverflow.com/questions/3338030/multiple-bash-traps-for-the-same-signal
+#===================================================================
+trap_add() {
+    trap_add_cmd=$1; shift || fatal "${FUNCNAME} usage error"
+    new_cmd=
+    for trap_add_name in "$@"; do
+        # Grab the currently defined trap commands for this trap
+        existing_cmd=`trap -p "${trap_add_name}" |  awk -F"'" '{print $2}'`
+
+        # Define default command
+        [ -z "${existing_cmd}" ] && existing_cmd="echo exiting @ `date`"
+
+        # Generate the new command
+        new_cmd="${trap_add_cmd};${existing_cmd}"
+
+        # Assign the test
+         trap   "${new_cmd}" "${trap_add_name}" || \
+                fatal "unable to add to trap ${trap_add_name}"
+    done
+}
+
+set -ex
+
+TAG=$(git rev-parse --short HEAD)
+BASE_IMAGE=quay.io/example/helm-app-operator:${TAG}
+MEMCACHED_IMAGE=quay.io/example/memcached-operator:${TAG}
+
+# switch to the "default" namespace if on openshift, to match the minikube test
+if which oc 2>/dev/null; then oc project default; fi
+
+# build operator base image
+docker build -t ${BASE_IMAGE} -f build/Dockerfile .
+
+# build a memcached operator
+pushd test
+pushd memcached-operator
+DIR1=$(pwd)
+
+mkdir chart && wget -qO- https://storage.googleapis.com/kubernetes-charts/memcached-2.3.1.tgz | tar -xzv --strip-components=1 -C ./chart
+trap_add 'rm -rf ${DIR1}/chart' EXIT
+
+docker build --build-arg BASE_IMAGE=${BASE_IMAGE} -t ${MEMCACHED_IMAGE} .
+sed "s|REPLACE_IMAGE|${MEMCACHED_IMAGE}|g" deploy/operator.yaml.tmpl > deploy/operator.yaml
+sed -i "s|Always|Never|g" deploy/operator.yaml
+trap_add 'rm ${DIR1}/deploy/operator.yaml' EXIT
+
+# deploy the operator
+kubectl create -f deploy/rbac.yaml
+trap_add 'kubectl delete -f ${DIR1}/deploy/rbac.yaml' EXIT
+
+kubectl create -f deploy/crd.yaml
+trap_add 'kubectl delete -f ${DIR1}/deploy/crd.yaml' EXIT
+
+kubectl create -f deploy/operator.yaml
+trap_add 'kubectl delete -f ${DIR1}/deploy/operator.yaml' EXIT
+
+
+# wait for operator pod to run
+if ! timeout 1m kubectl rollout status deployment/memcached-operator;
+then
+    kubectl describe deployment memcached-operator
+    kubectl logs deployment/memcached-operator
+    exit 1
+fi
+
+# create CR
+kubectl create -f deploy/cr.yaml
```

Comment:
Btw this is fine for now but just a heads up when we move this test into the SDK the project layout is a little different after the controller-runtime refactor:
https://github.com/operator-framework/operator-sdk-samples/tree/master/memcached-operator/deploy